src/Rest/RestApiBrowser.php

```diff
@@ -70,62 +110,48 @@ public function sendRequest($method, $url, $body = null, array $options = array(
         }
 
         if (null !== $this->responseStorage) {
-            $this->responseStorage->writeRawContent($this->response->getBody(true));
+            $this->responseStorage->writeRawContent($this->response->getBody()->getContents());
         }
     }
 
     /**
-     * @param string $name
-     * @param string $value
+     * @param string $method
+     * @param string $uri With or without host
+     * @param string|resource|array $body
      */
-    public function addRequestHeader($name, $value)
+    private function send($method, $uri, $body = null)
     {
-        if (isset($this->requestHeaders[$name])) {
-            if (!is_array($this->requestHeaders[$name])) {
-                $this->requestHeaders[$name] = array($this->requestHeaders[$name]);
-            }
-            $this->requestHeaders[$name][] = $value;
-        } else {
-            $this->requestHeaders[$name] = $value;
+        if (!$this->hasHost($uri)) {
+            $uri = rtrim($this->httpClient->getConfiguration()->getBaseUri(), '/') . '/' . ltrim($uri, '/');
         }
+        $stream = new Stream('php://memory', 'rw');
+        if ($body) {
+            $stream->write($body);
```

Comment:
I don't see any good reason for now to support ressource or array. Best option is to change phpdoc for now I guess.